src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,92 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { Post } from '../types/Post';
+import classNames from 'classnames';
+import { addComment } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post;
+  updateCommentList: Dispatch<SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  post,
+  updateCommentList,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
+  const [nameError, setNameError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [textError, setTextError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNameError('');
+    setName(event.target.value);
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEmailError('');
+    setEmail(event.target.value);
+  };
+
+  const handleCommentChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    setTextError('');
+    setText(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedName = name.trim();
+    const trimmedEmail = email.trim();
+    const trimmedText = text.trim();
+
+    let hasValidationErrors = false;
+
+    if (!trimmedName) {
+      hasValidationErrors = true;
+      setNameError('Name is required');
+    }
+
+    if (!trimmedEmail) {
+      hasValidationErrors = true;
+      setEmailError('Email is required');
```

Comment:
Consider adding a more robust email validation to ensure the email format is correct. Currently, it only checks if the email field is empty.